### ISSUE TITLE
Slow performance of VariableManager.get_vars()

### ISSUE THREAD

**Issue by phemmer** [ansible/ansible#74380]
### Summary

While attempting to diagnose performance issues of slow ansible runs, I discovered that a vast majority of the time is being spent in [`VariableManager.get_vars()`](https://github.com/ansible/ansible/blob/v2.10.8/lib/ansible/vars/manager.py#L146).

![image](https://user-images.githubusercontent.com/1826947/115716445-301acb80-a347-11eb-92ee-0188d25bf3fe.png)

ncalls | tottime | percall | cumtime | percall | filename:lineno(function)
-- | -- | -- | -- | -- | --
1 | 0.000742 | 0.000742 | 234.6 | 234.6 | ansible-playbook:22(<module>)
1 | 2.4e-05 | 2.4e-05 | 234.1 | 234.1 | playbook.py:69(run)
1 | 0.0001 | 0.0001 | 234 | 234 | playbook_executor.py:75(run)
1 | 0.3043 | 0.3043 | 233.4 | 233.4 | linear.py:199(run)
1 | 4.3e-05 | 4.3e-05 | 233.4 | 233.4 | task_queue_manager.py:209(run)
588/497 | 0.6874 | 0.001383 | 183.9 | 0.37 | manager.py:146(get_vars)
29511830/1089 | 36.66 | 0.03366 | 122.8 | 0.1127 | unsafe_proxy.py:116(wrap_var)
2460576/1089 | 9.84 | 0.009036 | 122.8 | 0.1127 | unsafe_proxy.py:100(_wrap_dict)
15957059/5444671 | 6.083 | 1.117e-06 | 115.8 | 2.126e-05 | unsafe_proxy.py:101(<genexpr>)
142517289/142517285 | 22.69 | 1.592e-07 | 80.05 | 5.617e-07 | ~:0(<isinstance>)
49959139 | 35.25 | 7.056e-07 | 57.35 | 1.148e-06 | abc.py:128(__instancecheck__)
544 | 3.671 | 0.006749 | 45.2 | 0.08309 | clean.py:118(clean_facts)
17 | 0.2317 | 0.01363 | 42.56 | 2.504 | __init__.py:796(_wait_on_pending_results)
34825 | 38.08 | 0.001094 | 38.08 | 0.001094 | ~:0(<time.sleep>)
1386477/1148 | 4.959 | 0.00432 | 29.82 | 0.02598 | clean.py:69(strip_internal_keys)

This seems like a major amount of overhead.

Can we get some optimizations to reduce this?

### Issue Type

Bug Report

### Component Name

vars

### Ansible Version

```console
$ ansible --version
ansible 2.10.8
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/etc/haproxy-ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/virtualenv/ansible-2.10-test/local/lib/python2.7/site-packages/ansible
  executable location = /usr/local/virtualenv/ansible-2.10-test/bin/ansible
  python version = 2.7.9 (default, Sep 25 2018, 20:42:16) [GCC 4.9.2]
```

### Configuration

```console
$ ansible-config dump --only-changed
ANSIBLE_PIPELINING(/etc/ansible/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/etc/ansible/ansible.cfg) = -o ControlMaster=auto -o ControlPersist=900s -o ConnectTimeout=5s -o ConnectionAttempts=3 -o PreferredAuthentications=publickey
ANSIBLE_SSH_CONTROL_PATH_DIR(/etc/ansible/ansible.cfg) = /var/lib/haproxy-ansible/cp
ANSIBLE_SSH_EXECUTABLE(/etc/ansible/ansible.cfg) = /usr/bin/ssh
ANSIBLE_SSH_RETRIES(/etc/ansible/ansible.cfg) = 3
ANY_ERRORS_FATAL(/etc/ansible/ansible.cfg) = True
CACHE_PLUGIN(/etc/ansible/ansible.cfg) = jsonfile
CACHE_PLUGIN_CONNECTION(/etc/ansible/ansible.cfg) = /var/lib/haproxy-ansible/jsonfile.d
CACHE_PLUGIN_TIMEOUT(/etc/ansible/ansible.cfg) = 604800
DEFAULT_CALLBACK_WHITELIST(/etc/ansible/ansible.cfg) = [u'profile_tasks', u'timer']
DEFAULT_FILTER_PLUGIN_PATH(/etc/ansible/ansible.cfg) = [u'/root/.ansible/plugins/filter', u'/etc/haproxy-ansible/plugins/filter', u'/usr/share/ansible/plugins/filter']
DEFAULT_FORKS(/etc/ansible/ansible.cfg) = 20
DEFAULT_GATHERING(/etc/ansible/ansible.cfg) = smart
DEFAULT_HOST_LIST(/etc/ansible/ansible.cfg) = [u'/etc/haproxy-ansible/inventory', u'/etc/ansible/inventory']
DEFAULT_LOCAL_TMP(/etc/ansible/ansible.cfg) = /var/lib/haproxy-ansible/tmp/ansible-local-18768CCe7hD
DEFAULT_LOG_PATH(/etc/ansible/ansible.cfg) = /var/log/haproxy-ansible.log
DEFAULT_MODULE_PATH(/etc/ansible/ansible.cfg) = [u'/root/.ansible/plugins/modules', u'/etc/haproxy-ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
DEFAULT_PRIVATE_KEY_FILE(/etc/ansible/ansible.cfg) = /root/.ssh/ansible-controller-id_rsa
DEFAULT_ROLES_PATH(/etc/ansible/ansible.cfg) = [u'/root/.ansible/roles', u'/etc/haproxy-ansible/roles', u'/usr/share/ansible/roles', u'/etc/ansible/roles']
DEFAULT_SCP_IF_SSH(/etc/ansible/ansible.cfg) = True
DEFAULT_STDOUT_CALLBACK(/etc/ansible/ansible.cfg) = unixy
DEFAULT_TIMEOUT(/etc/ansible/ansible.cfg) = 60
DEFAULT_TRANSPORT(/etc/ansible/ansible.cfg) = ssh
DEFAULT_VAULT_PASSWORD_FILE(/etc/ansible/ansible.cfg) = /etc/haproxy-ansible/vault-password
DISPLAY_SKIPPED_HOSTS(/etc/ansible/ansible.cfg) = False
HOST_KEY_CHECKING(/etc/ansible/ansible.cfg) = False
INVENTORY_UNPARSED_IS_FAILED(/etc/ansible/ansible.cfg) = True
RETRY_FILES_ENABLED(/etc/ansible/ansible.cfg) = False
```

### OS / Environment

Debian Jessie

### Steps to Reproduce

```
python -m cProfile -o /tmp/profile ansible-playbook ...
```

### Expected Results

Ansible should spend the majority of its time waiting on remote execution.

### Actual Results

```console
Ansible spent <20% of its time waiting on remote execution.
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
* [`lib/ansible/plugins/lookup/vars.py`](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/lookup/vars.py)

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by mkrizek** [comment]
Possible duplicate of https://github.com/ansible/ansible/issues/73654#issuecomment-784858199

**Comment by bcoca** [comment]
closing as dupe
